### DISCUSSION TITLE
Problem while implementing not gate with transistors

### DISCUSSION THREAD

**Discussion by me-nkr** [logisim-evolution/logisim-evolution#1898] [Q&A] [answered]
I have been trying to unload all the libraries and build stuff from the NAND gate only, but I can't unload the gates library since the NAND gate is in it. So I was trying to unload that library and create custom gates with transistors, and I ran into this problem.

I was tinkering with transistors trying to implement NOT gate and in these are the results

<img width="250" alt="notOff" src="https://github.com/logisim-evolution/logisim-evolution/assets/60778845/498ee90a-77b0-4a37-b17c-fbaa9418ea29">
<img width="197" alt="notOn" src="https://github.com/logisim-evolution/logisim-evolution/assets/60778845/968a7cd9-84a1-4366-93c8-89b81601f4f2">

when input is `1` the output is not `0`, why is this?
Can this be solved?

**Comment by davidhutchens** [comment, marked as answer]
You have your N-Type transistor facing South toward ground. Flip it to North and the circuit will work as expected. The little arrow in the transistor shows the information flow, not the current flow.

You are not required to use all the gates in the library. You can certainly build everything with NAND gates by just avoiding using the others. We provide transistors for the sole purpose of demonstrating how to build simple gates with them. We recommend not using them for larger designs, but they should work when used as specified.
